core/src/main/java/org/mule/transport/polling/PollingMessageSource.java

```diff
@@ -0,0 +1,421 @@
+import org.mule.api.execution.ExecutionCallback;
+import org.mule.api.execution.ExecutionTemplate;
+import org.mule.api.lifecycle.CreateException;
+import org.mule.api.lifecycle.Disposable;
+import org.mule.api.lifecycle.Initialisable;
+import org.mule.api.lifecycle.InitialisationException;
+import org.mule.api.lifecycle.Startable;
+import org.mule.api.lifecycle.Stoppable;
+import org.mule.api.processor.MessageProcessor;
+import org.mule.api.schedule.Scheduler;
+import org.mule.api.schedule.SchedulerFactory;
+import org.mule.api.source.MessageSource;
+import org.mule.config.i18n.CoreMessages;
+import org.mule.context.notification.ConnectorMessageNotification;
+import org.mule.execution.TransactionalErrorHandlingExecutionTemplate;
+import org.mule.transport.NullPayload;
+import org.mule.transport.TrackingWorkManager;
+import org.mule.util.StringUtils;
+import org.mule.util.concurrent.NamedThreadFactory;
+import org.mule.util.concurrent.ThreadNameHelper;
+
+import java.util.Map;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledThreadPoolExecutor;
+import java.util.concurrent.ThreadFactory;
+import java.util.concurrent.TimeUnit;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+//MULE-9139 - TODO verify if only one connector was created for all the polling elements in the config. If that's the case then we need to create a centralized service for the scheduling service
+/**
+ * <p>
+ * Polling {@link org.mule.api.source.MessageSource}.
+ * </p>
+ * <p>
+ * The {@link PollingMessageSource} is responsible of creating a {@link org.mule.api.schedule.Scheduler}
+ * at the initialization phase. This {@link org.mule.api.schedule.Scheduler} can be stopped/started and executed by using the {@link org.mule.api.registry.MuleRegistry}
+ * interface, this way users can manipulate poll from outside mule server.
+ * </p>
+ */
+public class PollingMessageSource implements MessageSource, FlowConstructAware, Startable, Stoppable, MuleContextAware, Initialisable, Disposable
+{
+
+    private static Logger logger = LoggerFactory.getLogger(PollingMessageSource.class);
+
+    /**
+     * <p>
+     * The Polling name identifier. Used to create the scheduler name
+     * </p>
+     */
+    public static final String POLLING_SCHEME = "polling";
+
+    /**
+     * <p>
+     * Format string for all the Polling Schedulers name.
+     * </p>
+     */
+    private static final String POLLING_SCHEDULER_NAME_FORMAT = POLLING_SCHEME + "://%s/%s";
+    private final SchedulerFactory<Runnable> schedulerFactory;
+
+    /**
+     * The {@link org.mule.api.schedule.Scheduler} instance used to execute the scheduled jobs
+     */
+    private Scheduler scheduler;
+
+    private MessageProcessor listener;
+
+    public static final long DEFAULT_POLL_FREQUENCY = 1000;
+    public static final TimeUnit DEFAULT_POLL_TIMEUNIT = TimeUnit.MILLISECONDS;
+
+    private long frequency = DEFAULT_POLL_FREQUENCY;
```

Comment:
This should go shouldn't it? Isn't this replaced via use of FixedFrequencyScheduler?  Also in the constructor and the getters/setters.